packages/components/src/DateTimePickers/InputDateTimePicker/date-extraction.js

```diff
@@ -380,8 +380,40 @@ function extractPartsFromTextInput(textInput, options) {
 	};
 }
 
+/**
+ * Extract parts (date, time, date/time, textInput) from a value with
+ * different possible types
+ * @param value {string | Date \ number}
+ * @param options {Object}
+ * @returns
+ *	{{
+ *		date: Date,
+ *		time: { hours: string, minutes: string, seconds: string },
+ *		datetime: Date,
+ *		textInput: string
+ * 	}}
+ */
+function extractParts(value, options) {
+	const typeOfValue = typeof value;
+	if (typeOfValue === 'number') {
+		return extractPartsFromDateTime(new Date(value), options);
+	} else if (typeOfValue === 'string') {
+		return extractPartsFromTextInput(value, options);
+	} else if (value instanceof Date) {
+		return extractPartsFromDateTime(value, options);
+	}
+
+	return {
```

Comment:
what about  `return null` instead of an invalid object? 
I got very bad experience with those kind of empty/invalid instances

Reply:
No, the values in the returned object are used in different locations.
For example, the date and time are used in the date picker, the textInput, in the input.
All those info will live in the InputDateTimePicker state. So we need the define each one of them for setState to replace the old values.